ios/architecture.md

```diff
@@ -1,34 +1,66 @@
+the interface and present the data received from view model. Also the view controller sends all actions to view model
+so it can handle them. The view model is the only layer, that handles state changes and the only place where the models
+are held. The models are retrieved from services which are injected using simple dependency injection container called
+the service holder.
 
-Services are mainly used for encapsulating some low-level frameworks like communication with server or persistence. They create higher-level API and enable simper testing by mocking them.
+Services are mainly used for encapsulating some low-level frameworks like communication with server or persistence.
+They create higher-level API and enable simpler testing by mocking them.
 
-The last layer in the MVVM-C architecture are coordinators (routers or navigators if you prefer). Coordinator is responsible for setting up the scene, navigating to it, back from it and navigating to other screens. In FuntastyKit there are some basic implementations of basic navigation patterns, namely push and modal. Of course, the concept is not limited to these basic patterns and is highly configurable.
+The last layer in the MVVM-C architecture are coordinators (routers or navigators if you prefer).
+Coordinator is responsible for setting up the scene, navigating to it, back from it and navigating to other screens.
+In FuntastyKit there are some basic implementations of basic navigation patterns, namely push and modal.
+Of course, the concept is not limited to these basic patterns and is highly configurable.
 
 ![MVVM-C architecture diagram](attachments/Figure.001.png)
 
 ## Dependency injection
 
-As a simple method for dependency injection a container called *service holder* is used. At the start of the application, first coordinator called `AppCoordinator` sets up one container and instantiates/adds the services to it, which are then available as singletons or multitons. Services can handle interaction with web APIs, persistence and pack other system APIs. Each service is primarily a protocol, so all the services are easily testable and mockable.
+As a simple method for dependency injection a container called *service holder* is used.
+At the start of the application, first coordinator called `AppCoordinator` sets up one container and instantiates/adds
+the services to it, which are then available as singletons or multitons. Services can handle interaction with web APIs,
+persistence and pack other system APIs. Each service is primarily a protocol, so all the services are easily testable
+and mockable.
 
-When the coordinator is navigating to other scenes, it passes the service holder along to them. The newly created coordinator then injects all the services the view model needs for handling its logic using the constructor injection.
+When the coordinator is navigating to other scenes, it passes the service holder along to them. The newly created
```

Comment:
Pripisal by som sem aj tu nevyhodu toho, ze ked chces z jednej sceny poslat nejaky model do druhej tak si to musis ulozit v coordinatoru a potom v configure nakonfigurovat ViewModel s tym modelom 🤔

Reply:
Nezdá se mi, že je to tak zásadní věc, že by bylo nutné klást jí takový důraz. Za mně je t ospíš vlastnost než nevýhoda. Co myslíš?

Reply:
Az take zasadne to neni mas pravdu ale kedze to tu je pekne popisana ta architektura tak jedna veticka by nicomu neuskodila si myslim. Napr. ja si pamatam, ze ked som zacinal tak som sa pytal Patrika ci to vzdy musim takto robit pokial chcem preniest nejaky model zo sceny A do sceny B.

Reply:
Máš pravdu, dopíšu.

Reply:
Dopsáno.